TechAndCheck/zenodotus
# Remove Tailwind’s Node package and related pieces

This PR removes the Tailwind Node package and any configuration or build stuff related to it. It wasn't being intentionally used anyway; we now use the Tailwind CSS gem and are moving toward a Webpacker-less future (#188).

I wish this PR could be chopped up into more small commits rather than one big one, but subscribing to a loose theory of individual commits not being knowingly broken required doing it all as one commit. The individual steps taken were:

1. The Tailwind CSS npm package was removed from `package.json`, and the lockfile updated accordingly.
2. The vestigial config files specific to that package (i.e., unused by the Tailwind CSS gem) were removed, along with an unused SCSS file that references more Tailwind.
3. With the npm package gone, the Yarn `build:css` task will fail (unless I do something like fill it with a noop), so that’s removed.
4. With the Yarn scripts gone, the `cssbundling-rails` gem complains during asset compilation, so that gem is commented out and the lockfile updated accordingly. (We don't want it anyway.)

It also includes a change to the README that instructs new devs to precompile assets, to get around some weirdness explained in the description of 9491976a61e837e99900f7d85e6c791f4b4754d3.

### Testing instructions:

1. `yarn install`
2. `bundle install`
3. Delete your `{repo}/public/assets` and  `{repo}/assets/builds` directories
4. `rails assets:precompile` ← this should actually be a first-run development step at all times, will add to README
6. `./bin/dev`
7. Check out the site, click around. If it's not a CSS-less void, all is well!

Closes #187
